src/test/run-pass/array-slice-vec/vec-matching-autoslice.stderr

```diff
@@ -0,0 +1,8 @@
+warning: unreachable pattern
+  --> $DIR/vec-matching-autoslice.rs:31:9
+   |
+LL |         ([_, _], _) => panic!(),
```

Comment:
Can you clarify why this is unreachable?

With a value of form `(_, _, 1.0)` it is not covered by the second arm, with `([(2, _), (_, true)], 1.0)` it is also not covered by the first; it is however covered by the last arm... Is this specific to floats somehow?

If you change the value this will also segfault (so its unsound): https://play.rust-lang.org/?version=nightly&mode=debug&edition=2015&gist=b0d6337036041d489f967caca6c83602

Assuming this is a bug, when fixing this, add a test file with `#![deny(unreachable_patterns)]` and the contents of the `.rs` file to ensure that this wont't happen.